[VatsalVitosh] iP

VatsalVitosh opened the pull request:
# Lia

> "You left your fridge open, somebody just took a sandwich" - Kanye West ([source](https://genius.com/8663294))

Lia helps you keep track of everything that you need to do. It's,

* Just like texting a friend
* Except that it's not
* Unless your friend's a MIPS processor
* Whose name is Lia
* Then I guess it's kind of like texting your friend

All you need to do is:

1. Download it.
2. Double-click it.
3. Add your tasks.
4. Develop existential dread because of all the work you're drowning in through the incessant marching of time and hope that life will eventually work out before the day of reckoning comes near.
5. And let it manage your tasks for you 😃

It's also **FREE** ~~(anti-capitalist moment)~~

Features:

- [ ] It's essentially just this
- [x] A list you can mark 
- [ ] And unmark
- [x] Its UI is honestly even worse than _this_ list
- [x] Please use it

Anyway, here's some `java code`:

```ruby
public class PublicClass {
    public static void main(String[] args) {
        System.out.println("Process finished with exit code 1")
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

VatsalVitosh committed:
Level 0 - Rename, Greet, Exit

VatsalVitosh committed:
Level 1 - Echo

VatsalVitosh committed:
Level 2 - Add, List

VatsalVitosh committed:
Level 3 - Mark as Done

VatsalVitosh committed:
Level 4 - ToDo, Event, Deadline

VatsalVitosh committed:
Level 5 - Handle Errors

VatsalVitosh committed:
Level 6 - Delete

VatsalVitosh committed:
A-TextUiTesting: Automated Text UI Testing

VatsalVitosh committed:
Update classes to include getters and setters

VatsalVitosh committed:
Update .gitignore to include /data/ directory

The /data/ directory contains Lia.txt which stores user specific data.

VatsalVitosh committed:
Implement ability to load/store data from/to hard-disk

VatsalVitosh committed:
Add ability to understand deadline dates in yyyy-MM-dd format and print back in MMM dd yyyy format

VatsalVitosh committed:
Merge branch 'branch-level-8'

VatsalVitosh committed:
Make program more OOP

VatsalVitosh committed:
Divide classes into packages

jjchee77 commented on src/main/java/lia/Storage.java:
```diff
@@ -0,0 +1,86 @@
+                    String description = parts[2];
+
+                    switch (type) {
+                        case "T":
+                            tasks.add(new Todo(description, isDone));
+                            break;
+                        case "D":
+                            String by = parts[3];
+                            tasks.add(new Deadline(description, by, isDone));
+                            break;
+                        case "E":
+                            String start = parts[3];
+                            String end = parts[4];
+                            tasks.add(new Event(description, start, end, isDone));
+                            break;
```
Don't think the case statements should be indented.

jjchee77 commented on src/main/java/lia/Storage.java:
```diff
@@ -0,0 +1,86 @@
+                        taskSave = "T | " + (task.isDone() ? "1" : "0") + " | " + task.getDescription();
+                    } else if (task instanceof Deadline) {
+                        taskSave = "D | " + (task.isDone() ? "1" : "0") + " | " +
+                                task.getDescription() + " | " + ((Deadline) task).getDate();
```
Perhaps the "+" should be on the next line.

jjchee77 commented on src/main/java/lia/Storage.java:
```diff
@@ -0,0 +1,86 @@
+                            String end = parts[4];
+                            tasks.add(new Event(description, start, end, isDone));
+                            break;
+                    }
```
Might want to add a default clause.

jjchee77 commented on src/main/java/lia/TaskList.java:
```diff
@@ -0,0 +1,65 @@
+        return tasks.get(pos);
+    }
+
+    public void markTaskAsDone(int pos){
```
The opening parenthesis could be preceded with a whitespace. There are also other occurrences.

jjchee77 reviewed (commented):
Mostly conforms to coding standard :) Just some nits that could be addressed

VatsalVitosh committed:
Merge branch 'add-gradle-support'

VatsalVitosh committed:
Set up Gradle so the project can be run using Gradle

VatsalVitosh committed:
Add JUnit tests for Parser,java and TaskList.java classes

VatsalVitosh committed:
Add JavaDoc comments to the code for all non-private classes and methods excluding getters and setters

VatsalVitosh committed:
Tweak the code to comply with a coding standard

VatsalVitosh committed:
Give users a way to find a task by using the find command

VatsalVitosh committed:
Merge branch 'branch-Level-9'

VatsalVitosh committed:
Update code to adhere to coding standard

VatsalVitosh committed:
Merge branch 'branch-A-CodingStandard'

VatsalVitosh committed:
Add JavaDoc headers for methods

VatsalVitosh committed:
Merge branch 'branch-A-JavaDoc'

VatsalVitosh committed:
Use CheckStyle to detect coding style violations

VatsalVitosh committed:
Merge branch 'branch-A-CheckStyle'

brennalaurentan commented on src/test/java/lia/ParserTest.java:
```diff
@@ -0,0 +1,51 @@
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class ParserTest {
```
Good implementation of test cases for testing!

brennalaurentan commented on src/main/java/lia/Storage.java:
```diff
@@ -0,0 +1,108 @@
+                    boolean isDone = parts[1].equals("1");
+                    String description = parts[2];
+
+                    switch (type) {
```
Might want to remove indentation for switch clauses (as per the coding conventions mentioned in this course)

brennalaurentan commented on src/main/java/lia/Storage.java:
```diff
@@ -0,0 +1,108 @@
+                for (Task task : tasks) {
+                    String taskSave;
+                    if (task instanceof Todo) {
+                        taskSave = "T | " + (task.isDone() ? "1" : "0") + " | " + task.getDescription();
```
Good use of ternary operators instead of if-else to shorten code!

brennalaurentan reviewed (commented):
LGTM! Nice task manager you've got here. Just a few comments below.

VatsalVitosh committed:
Fix indentation for switch case statements according to coding standard

reetmitra commented on src/main/java/lia/Deadline.java:
```diff
@@ -0,0 +1,44 @@
+        return "[" + getTaskIcon() + "]" + "[" + getStatusIcon() + "] " + getDescription()
+                + " (by: " + date.format(DateTimeFormatter.ofPattern("MMM dd yyyy")) + ")";
+    }
+}
```
JavaDoc comments are well written, gives a good explanation of each method.

reetmitra commented on src/main/java/lia/Event.java:
```diff
@@ -0,0 +1,43 @@
+        this.end = end;
+    }
+
+    public String getStart() {
```
Maybe consider renaming to getStartTime for easier readability.

reetmitra commented on src/main/java/lia/Event.java:
```diff
@@ -0,0 +1,43 @@
+        return start;
+    }
+
+    public String getEnd() {
```
Similarly, getEndTime.

reetmitra commented on src/main/java/lia/Parser.java:
```diff
@@ -0,0 +1,184 @@
+        } catch (LiaException e) {
+            System.out.println(e.getMessage());
+        }
+    }
```
Maybe try using a switch case if that's possible?

reetmitra commented on src/main/java/lia/Storage.java:
```diff
@@ -0,0 +1,108 @@
+                    } else if (task instanceof Event) {
+                        taskSave = "E | " + (task.isDone() ? "1" : "0") + " | "
+                                + task.getDescription() + " | " + ((Event) task).getStart()
+                                + " | " + ((Event) task).getEnd();
```
Ternary operators have been used well to keep the code short and clean!

reetmitra reviewed (commented):
Overall, well written code with not much coding standard violations. JavaDocs are written quite well, explaining each method properly. Maybe try implementing some switch cases if that helps. Otherwise looks quite well structured.

VatsalVitosh committed:
Add JavaFX GUI

VatsalVitosh committed:
Add GUI to make Lia look like a text messaging app

VatsalVitosh committed:
Merge branch 'branch-Level-10'

VatsalVitosh committed:
Fix JavaDoc error

VatsalVitosh committed:
Edit build.gradle to be able to launch jar file

VatsalVitosh committed:
Add assertions to document important assumptions within the code

Assert that positions within the TaskList when retrieving and deleting are valid.
Assert that positions within the TaskList when marking tasks are valid.
Assert that keyword when using find command should not be empty.

VatsalVitosh committed:
Make edits to potentially improve code quality

Edit code so that when catching LiaExceptions, the system logs the error message by printing the stack trace.
Add exception handling to getResponse method to account for invalid inputs.

VatsalVitosh committed:
Remove git tracking for test files

Users do not require test files in the program.

VatsalVitosh committed:
Merge pull request #2 from VatsalVitosh/branch-A-Assertions

Add assertions to document important assumptions within the code

VatsalVitosh committed:
Merge branch 'master' of https://github.com/VatsalVitosh/ip

VatsalVitosh committed:
Merge branch 'master' into branch-A-CodeQuality

VatsalVitosh committed:
Merge pull request #3 from VatsalVitosh/branch-A-CodeQuality

Edit to improve code quality

VatsalVitosh committed:
Merge branch 'master' of https://github.com/VatsalVitosh/ip

VatsalVitosh committed:
Add ability for user to be able to mark tasks as important

VatsalVitosh committed:
Merge tag 'BCD-Extension'

VatsalVitosh committed:
Update README.md with user guide and add product screenshot

VatsalVitosh committed:
Fix filepath for Ui picture for github pages

VatsalVitosh committed:
Add alt text for screenshot
